File: Dart/src/Game.dart
class Game {
  final Board _board;
  final Ant _ant;
  
  Game(Ant this._ant, Board this._board);
  
  void tick() {
    if(_board.isBlack(_ant.position))
      _ant.turn(Rotation.LEFT);
    else
      _ant.turn(Rotation.RIGHT);
    _board.flipColor(_ant.position);
    _ant.move();
  }
}

File: Kotlin/src/Game.kt
import java.util.Collection
import java.util.Set
import java.util.HashSet

class Game(val board: Board, val ant: Ant){
    fun tick() {
        if (board.isBlack(ant.position))
            ant.turn(Turn.Left)
        else
            ant.turn(Turn.Right)
        board.flip(ant.position)
        ant.move()
    }
}

